utils/swift_build_support/swift_build_support/products/playgroundsupport.py

```diff
@@ -0,0 +1,126 @@
+# swift_build_support/products/playgroundsupport.py -------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+import re
+
+from . import product
+from .. import shell
+from .. import targets
+
+
+def get_os_spelling(os):
+    return {
+        'macosx': 'macOS',
+        'iphonesimulator': 'iOS',
+        'appletvsimulator': 'tvOS',
+    }[os]
+
+
+class PlaygroundSupport(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        return "swift-xcode-playground-support"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return self.args.build_playgroundsupport
+
+    def build(self, host_target):
+        root = os.path.dirname(os.path.dirname(self.toolchain.swiftc))
+        swift_lib_dir = os.path.join(root, 'lib', 'swift')
+        (host_os, host_arch) = host_target.split('-')
+
+        with shell.pushd(self.source_dir):
+            shell.call([
+                "xcodebuild",
+                "-configuration", self.args.build_variant,
+                "-workspace", "swift-xcode-playground-support.xcworkspace",
+                "-scheme", "BuildScript-{}".format(get_os_spelling(host_os)),
+                "-sdk", host_os,
+                "-arch", host_arch,
+                "-derivedDataPath", os.path.join(self.build_dir, "DerivedData"),
+                "SWIFT_EXEC={}".format(self.toolchain.swiftc),
+                "SWIFT_LIBRARY_PATH={}/$(PLATFORM_NAME)".format(swift_lib_dir),
+                "ONLY_ACTIVE_ARCH=NO",
+            ])
+
+    def should_test(self, host_target):
+        return re.match('macosx', host_target) and \
+            self.args.test_playgroundsupport
+
+    def test(self, host_target):
+        root = os.path.dirname(os.path.dirname(self.toolchain.swiftc))
+        swift_lib_dir = os.path.join(root, 'lib', 'swift')
+        (host_os, host_arch) = host_target.split('-')
+
+        with shell.pushd(self.source_dir):
+            shell.call([
```

Comment:
Since this new structure does the building and testing one right after each other, you can replace these two separate `xcodebuild` calls with just one; pass `test` instead of `build-for-testing`/`test-without-building`

Reply:
😮that is way better, I didn't know about that, thanks!  Let's wait for the toolchain build and I'll update it for that simplification.

Reply:
Sounds good! (And yeah, it is way simpler, I went the other way so that building would build everything needed for the tests in build-script-impl but I think this just makes more sense.)

Reply:
Looks good now!